File: 083/remove-duplicates-from-sorted-list.go
type ListNode struct {
	Val  int
	Next *ListNode
}

func deleteDuplicates(head *ListNode) *ListNode {
	curr := head
	ln := head

	for ln != nil {
		if curr.Val < ln.Val {
			curr.Next = ln
			curr = curr.Next
		}
		ln = ln.Next
	}

	if curr != nil {
		curr.Next = nil
	}
	return head
}

File: 083/remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution:
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        curr = head
        ln = head
        while ln is not None:
            if curr.val < ln.val:
                curr.next = ln
                curr = curr.next

            ln = ln.next

        if curr is not None:
            curr.next = None

        return head
